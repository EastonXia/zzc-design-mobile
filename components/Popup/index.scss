@import '../style/base.scss';
$cls: '.zds-popup';
#{$cls} {
    position: fixed;
    top: 0;
    left: 0;
    z-index: 1001;
    height: 100%;
    width: 100%;
    @include flex();
    
    #{$cls}-mask {
        background: rgba(23, 22, 41, .7);
        height: 100%;
        width: 100%;
        top: 0;
        left: 0;
        position: absolute;
        z-index: 999;
    }
    #{$cls}-box {
        position: absolute;
        background: $c_fff;
        z-index: 1001;
        width: 100%;
        &#{$cls}-bottom{
            border-top-left-radius: 10px;
            border-top-right-radius: 10px;
            bottom:0;
            left: 0;
        }
        &#{$cls}-top{
            top:0;
            left: 0;
            border-bottom-left-radius: 10px;
            border-bottom-right-radius: 10px;
        }
        &#{$cls}-left{
            top:0;
            left: 0;
            bottom: 0;
            width: auto;
            border-top-right-radius: 10px;
            border-bottom-right-radius: 10px;
        }
        &#{$cls}-right{
            top:0;
            right: 0;
            bottom: 0;
            width: auto;
            border-top-left-radius: 10px;
            border-bottom-left-radius: 10px;
        }
    }
}

