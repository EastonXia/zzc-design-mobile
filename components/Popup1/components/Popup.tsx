import React from 'react';
import Dialog from '../../Dialog';
import animateConfig from '../../_util/animateConfig';
import config from '../../_util/config';
import { PopupProps } from '../porpsType';


export default class Popup extends React.PureComponent<PopupProps> {
    constructor( props ) {
        super( props );
    }
    static defaultProps = {
        prefixCls: `${config.cls}-popup`,
        visible: false,
        style: {},
        maskStyle: {},
        maskClose: false,
        onClose () { },
        transparent: false,
        direction: 'bottom',
        preRender: false
    }

    state = {
        showPopup: this.props.visible,
        lastShowPopup: this.props.visible,
        isClose: false,
        animationTypeList: animateConfig
    }

    private boxShowed: boolean = false;
    private rendered: boolean = false;

    componentWillReceiveProps ( nextProps ) {
        if ( nextProps.visible ) {
            this.setState( {
                lastShowPopup: true,
                showPopup: true
            } );
        } else if ( this.state.lastShowPopup && !nextProps.visible ) {
            this.setState( {
                isClose: true
            } );
        }
    }

    // 获取指定样式
    getAnimationClass ( direction: any ): any {
        const animation = this.state.animationTypeList[`${config.cls}-slide-${direction}`];
        if ( animation ) {
            return animation;
        }
        return direction ? direction : null;
    }

    boxAnimated( type: string ): void {
        this.boxShowed = type == 'enter';
        if ( this.boxShowed ) {
            this.renderCallback();
        }
    }
    maskAnimated(): void {
        if ( this.boxShowed  ) {
            this.renderCallback();
        }
    }

    renderCallback(): void {
        !this.rendered && this.props.renderCallback && this.props.renderCallback();
    }

    render () {
        const { visible, children, prefixCls, direction, onClose, preRender } = this.props;
        const { isClose, showPopup } = this.state;
        // 如果使用了预渲染模式的话，就直接渲染出Dialog
        if ( showPopup || (!isClose && preRender) ) {
            return (
                <Dialog
                    preRender={preRender}
                    maskTransitionName={`${config.cls}-fade`}
                    transitionName={this.getAnimationClass( direction )}
                    prefixCls={prefixCls}
                    boxClassName={`${prefixCls}-${direction}`}
                    visible={visible}
                    boxAnimated={( type ) => { this.boxAnimated( type ); }}
                    maskAnimated={() => { this.maskAnimated(); }}
                    closeCallback={() => {
                        this.setState( { showPopup: false } );
                        onClose && onClose();
                    }}
                    {...this.props}
                >
                    {children}
                </Dialog>
            );
        }
        return null;
    }
}