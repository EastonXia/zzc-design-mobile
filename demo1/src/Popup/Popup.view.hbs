<!DOCTYPE>
<html>
<head>
    <title>Popup</title>
    {{> '../../template/header.hbs'}}
    <link rel="stylesheet" href="../css/Popup.css">
<body>
    
<div id="root">

</div>
{{> '../../template/footer.hbs'}}
<script src="../js/Popup.js"></script>

</body>
</html>
