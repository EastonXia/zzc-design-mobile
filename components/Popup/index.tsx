import React, { PureComponent } from 'react';
import Error from '../_util/Error';
import Popup from './components/Popup';
import { PopupProps } from './porpsType';
import './index.scss';

export default class PopupEntrance extends PureComponent<PopupProps, any> {
    render() {
        return (
            <Error componentName='Popup'>
                <Popup {...this.props} />
            </Error>
        );
    }
}