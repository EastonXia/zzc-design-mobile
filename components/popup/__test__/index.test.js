import React from 'react';
import {render} from 'enzyme';
import {renderToJson} from 'enzyme-to-json';
import Popup from '../index.tsx';
test( 'Modal', () => {
    const wrapper = render(
        <div>
            <Popup
                maskClose={true}
                visible={true}
            >
                <h1>123</h1>
            </Popup>
            <Popup
                maskClose
                preRender
                visible={false}
                onClose={() => {}}
            >
                <div className='card-box2'>
                    <p>文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容</p>
                    <p>文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容</p>
                    <p>文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容</p>
                </div>
            </Popup>
        </div>
    );
    expect( renderToJson( wrapper ) ).toMatchSnapshot();
} );

