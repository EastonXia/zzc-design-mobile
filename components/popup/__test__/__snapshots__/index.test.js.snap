// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal 1`] = `
<div>
  <div
    class="zds-popup"
  >
    <div
      class="zds-popup-mask"
    />
    <div
      class="zds-popup-box zds-popup-bottom"
    >
      <h1>
        123
      </h1>
    </div>
  </div>
  <div
    class="pre-render-box"
    style="display:none"
  >
    <div
      class="zds-popup"
    >
      <div
        class="zds-popup-mask"
      />
      <div
        class="zds-popup-box zds-popup-bottom"
      >
        <div
          class="card-box2"
        >
          <p>
            文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容
          </p>
          <p>
            文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容
          </p>
          <p>
            文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
